File: Day264/minimumNumberOfOperationToMakeArrayContinuous.java
package Day264;

import java.util.Arrays;

public class minimumNumberOfOperationToMakeArrayContinuous {
    
      public static int minimumOperation(int arr[]) {
        Arrays.sort(arr);
        int answer = Integer.MAX_VALUE;
        int j = 1, temp = 0;
        int tempArr[] = new int[arr.length];
        for (int i=0; i<arr.length; i++) {
            while (j<arr.length && arr[j] < arr[i] + arr.length-1) {
                if (arr[j] == arr[j-1]) {
                    temp++;

                }
                tempArr[j] = temp;
                temp++;
            }
            answer = Math.min(answer, i+(arr.length-j)+temp-tempArr[i]);
        }
        return answer;
    }

    public static void main(String[] args) {
        int arr1[] = {4,2,5,3};
        int arr2[] = {1,2,3,4,5,6};
        int arr3[] = {1,10,100,1000};
        System.out.println(minimumOperation(arr1));
        System.out.println(minimumOperation(arr2));
        System.out.println(minimumOperation(arr3));
    }
}

File: Day264/minimumNumberOfOperationToMakeArrayContinuous.go
package main

import (
	"fmt"
	"math"
	"sort"
)

func minimumOperation(nums []int) int {
	answer := math.MaxInt
	sort.Ints(nums)
	j, temp := 1, 0
	tempArr := make([]int, len(nums))
	for i:=0; i<len(nums); i++ {
		for j<len(nums) && nums[j]<=nums[i]+len(nums)-1 {
			if nums[j] == nums[j-1] {
				temp++
			}
			tempArr[j] = temp;
			j++;
		}
		answer = int(math.Min(float64(answer), float64(i + (len(nums)-j) + temp - tempArr[i])))
	}
	return answer
}

func main() {
	arr1 := []int{4,2,5,3}
	arr2 := []int{1,2,3,4,5,6}
	arr3 := []int{1,10,100,1000}
	fmt.Println(minimumOperation(arr1))
	fmt.Println(minimumOperation(arr2))
	fmt.Println(minimumOperation(arr3))
}

